lib/elixir/lib/gen_server.ex

```diff
@@ -159,6 +159,71 @@ defmodule GenServer do
   the GenServer callbacks as doing so will cause the GenServer to misbehave.
   If you want to receive custom messages, always receive them in `handle_info/2`.
 
+  ## Debugging with the `:sys` module
+
+  GenServers, as all [special processes](http://erlang.org/doc/design_principles/spec_proc.html)
+  can be debugged using the `:sys` module. Through various hooks, this module
+  allows developers to introspect into the state of the process and trace
+  system events that happen during its execution, such as received messages,
+  sent replies and state changes.
+
+  Let's explore the basic functions from the `:sys` module used for debugging:
+
+    * [`:sys.get_state/2`](http://erlang.org/doc/man/sys.html#get_state-2) -
+      allows us to retrieve the state of the process. In case of
+      a GenServer process, it will be the callback module state, as
+      passed into the callback functions as last argument.
+    * [`:sys.get_status/2`](http://erlang.org/doc/man/sys.html#get_status-2) -
+      allows to retrieve the status of the process. This status includes
+      process dictionary, if the process is running or is suspended,
+      parent PID, current debug options, and state of the behaviour module,
+      that includes the callback module state (as returned by `:sys.get_state/2`).
+      It's possible to change how this status is represented by defining
+      the optional `format_status/2` callback.
```

Comment:
Does this link to the callback? Maybe we need to use `c:GenServer.format_status/2`.